File: B_10828.py
import sys

n = int(input())
stack = []

for i in range(n):
    s = sys.stdin.readline().split()

    if s[0] == 'push' :
        stack.append(s[1])

    elif s[0] == 'pop':
        if not stack:
            print(-1)
        else:
            print(stack.pop())

    elif s[0] == 'size':
        print(len(stack))
    
    elif s[0] == 'empty':
        if stack:
            print(0)
        else:
            print(1)
    
    elif s[0] == 'top':
        if not stack:
            print(-1)
        else:
            print(stack[-1])

File: java/src/main/java/B_10828.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Stack;
import java.util.StringTokenizer;

public class B_10828 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();

        int n = Integer.parseInt(br.readLine());
        Stack<Integer> stack = new Stack<Integer>();

        for(int i = 0; i < n; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            String str = st.nextToken();

            if(str.equals("push")) {
                stack.push(Integer.parseInt(st.nextToken()));
            } else if(str.equals("pop")) {
                if(stack.empty()) sb.append(-1).append("\n");
                else sb.append(stack.pop()).append("\n");
            } else if(str.equals("size")) {
                sb.append(stack.size()).append("\n");
            } else if(str.equals("empty")) {
                if(stack.empty()) sb.append(1).append("\n");
                else sb.append(0).append("\n");
            } else {
                if(stack.empty()) sb.append(-1).append("\n");
                else sb.append(stack.peek()).append("\n");
            }
        }
        System.out.println(sb);
    }
}
